indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -1950,6 +1950,58 @@ public Boolean apply(SeekableStreamIndexTaskRunner.Status status)
     // make sure the checkpoints are consistent with each other and with the metadata store
 
     verifyAndMergeCheckpoints(taskGroupsToVerify.values());
+
+    // A pause from the previous Overlord's supervisor, immediately before leader change,
+    // can lead to tasks being in a state where they are active but do not read.
+    resumeAllActivelyReadingTasks();
```

Comment:
how frequently is `discoverTasks` going to be called? This wouldn't just be called after leader re-election, right? There is also code to get the status of all tasks and then doing an action based on what the current status is. can the resuming go inside that block of code itself? So if the status returned is paused, only then we will be calling `resumeAsync` instead of doing it for each task.

Reply:
While discoverTasks is called in every RunNotice, there is a check within this method to resume only during the first run or return otherwise.
It is called here as the pendingCompletionTaskGroups are determined here using the status i.e whether it is publishing or not. These tasks are not affected, and only the activelyReadingTaskGroups are resumed